### DISCUSSION TITLE
SOLWEIG Analyzer - Python error "daymean" variable referenced before assignment

### DISCUSSION THREAD

**Discussion by husamah7** [UMEP-dev/UMEP#595] [Q&A]
Hello, 
I am trying to complete the Thermal Comfort - Introduction to SOLWEIG Tutorial available [here](https://umep-docs.readthedocs.io/projects/tutorial/en/latest/Tutorials/IntroductionToSolweig.html#) 
And when I am doing the very last step in this tutorial - trying to create a new threshold map - I get this error message: 

Traceback (most recent call last):
File "/Users/husamalahmadieh/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/processing_umep/postprocessor/solwieganalyzer_algorithm.py", line 233, in processAlgorithm
daymean = daymean / index
UnboundLocalError: local variable 'daymean' referenced before assignment

I am not sure if I did something wrong, but I did follow the steps how they're outlined in the tutorial. Do you have any suggestion as to what I can do?

Thanks!!

**Comment by biglimp** [comment]
Just quickly tried and did not get the same error. I am sitting on a Windows machine. It looks like you might be on something else? The other thing you can check is that you have files in your output folder form the SOLWEIG run.

**Reply by husamah7** [reply]
Hey, thanks for the reply!!

I'm using a Macbook Air with QGIS version 3.28.12-Firenze. 
Just tried changing my output folder (created a new one) and still got the same error.

**Reply by biglimp** [reply]
Might be a Mac issue. Could you zip and attach the output folder from the SOLWEIG run for me to have a closer look.

**Reply by husamah7** [reply]
Hello,

I have attached the folder I have been working with including the QGIS file. Thank you for the help!!
[Tutorial for Github.zip](https://github.com/UMEP-dev/UMEP/files/14744510/Tutorial.for.Github.zip)

**Reply by biglimp** [reply]
I see that you have no relevant data in **output including veg** and **output including veg and landcover**. If you use any of those two as **Path to SOLWEIG output folder**, the error that you get will be present. Something must have gone wrong when you ran SOLWEIG for those two cases. If you instead use your folder **output**, then you get an output raster, yes?

**Reply by husamah7** [reply]
Unfortunately not :/ I still get the same error. 
I will try doing the whole tutorial again, maybe then it will work.

Thank you for your help!!
